source/extensions/quic_listeners/quiche/platform/quic_bug_tracker_impl.h

```diff
@@ -6,9 +6,9 @@
 // consumed or referenced directly by other Envoy code. It serves purely as a
 // porting layer for QUICHE.
 
-#include "quiche/quic/platform/api/quic_logging.h"
+#include "extensions/quic_listeners/quiche/platform/quic_logging_impl.h"
 
-#define QUIC_BUG_IMPL QUIC_LOG(DFATAL)
-#define QUIC_BUG_IF_IMPL(condition) QUIC_LOG_IF(DFATAL, condition)
-#define QUIC_PEER_BUG_IMPL QUIC_LOG(ERROR)
-#define QUIC_PEER_BUG_IF_IMPL(condition) QUIC_LOG_IF(ERROR, condition)
+#define QUIC_BUG_IMPL QUIC_LOG_IMPL(DFATAL)
```

Comment:
preexisted, but doesn't QUIC_BUG work based on the exponential back-off, and could cause serious perf problems if we don't do that?
Should we get a TODO to get around to that work?

Reply:
QUIC_BUG has the exponential back off in google3 but not Chromium. It, like other error logs, could cause serious perf problems if it happens too frequently.

TODO added to implement exponential back off in envoy.

Reply:
yeah, per-packet logging server side scales less well than client side :-)   thanks for the TODO